iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/helpers/Tools.java

```diff
@@ -124,6 +124,115 @@ public static boolean isCause(Class<? extends Throwable> possibleExceptionCause,
         return possibleExceptionCause.isInstance(exceptionToSearch) || (exceptionToSearch != null && isCause(possibleExceptionCause, exceptionToSearch.getCause()));
     }
 
+    private static RegistryManager registryManager;
+
+    public static RegistryManager getRegistyManager(String iotHubConnectionString) throws IOException
+    {
+        if (registryManager == null)
+        {
+            RegistryManagerOptions options = RegistryManagerOptions.builder().httpReadTimeout(HTTP_READ_TIMEOUT).build();
+            registryManager = RegistryManager.createFromConnectionString(iotHubConnectionString, options);
+        }
+
+        return registryManager;
+    }
+
+    public static TestDeviceIdentity getTestDevice(String iotHubConnectionString, IotHubClientProtocol protocol, AuthenticationType authenticationType) throws URISyntaxException, IOException, IotHubException, GeneralSecurityException, InterruptedException
+    {
+        RegistryManager registryManager = getRegistyManager(iotHubConnectionString);
+        if (authenticationType == SAS)
+        {
+            String deviceId = "java-file-upload-e2e-test-".concat(UUID.randomUUID().toString());
```

Comment:
Is this helper method only for file-upload tests?

Reply:
It is not. This prefix is just a copy paste error, so I'll get rid of it